src/Distribution/Server/Features/UserNotify.hs

```diff
@@ -817,18 +832,6 @@ userNotifyFeature serverEnv@ServerEnv{serverBaseURI, serverCron}
       Map.mapKeys (, pid) <$>
         getUserNotificationsOnRelease (queryUserGroup . maintainersGroup) idx revIdx queryGetUserNotifyPref pid
 
-    describeRevision users earlier now pkg
-      | pkgNumRevisions pkg <= 1 =
-          EmailContentParagraph $
-            "Package upload, " <> renderPkgLink (pkgInfoId pkg) <> ", by " <>
-            formatTimeUser users (pkgLatestUploadTime pkg) (pkgLatestUploadUser pkg)
-      | otherwise =
-          EmailContentParagraph ("Package metadata revision(s), " <> renderPkgLink (pkgInfoId pkg) <> ":")
-          <> EmailContentList (map (uncurry (formatTimeUser users) . snd) recentRevs)
-        where
-           revs = reverse $ Vec.toList (pkgMetadataRevisions pkg)
-           recentRevs = filter ((\x -> x > earlier && x <= now) . fst . snd) revs
```

Comment:
Previously, the logic was broken up as:
1. `collectRevisionsAndUploads`: Filter packages if they have recent revisions
2. `genRevUploadList`: Filter packages if user should be notified
3. `describeRevision`: Find the recent revisions + render them into message

Now, finding the recent revisions is moved into `genRevUploadList` so that `renderNotifyNewRevision` can just be responsible for rendering into an email.